app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,73 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
```

Comment:
You should define your own current_user and authenticate_user! helper methods instead of using the ones from devise gem. Define them in the `application_controller.rb`. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_post_requirement.png) from Odin.